cmd/libsnap-confine-private/cleanup-funcs.h

```diff
@@ -27,6 +27,10 @@
 #include <sys/types.h>
 #include <dirent.h>
 
+// SC_CLEANUP will run the given cleanup function when the variable next
+// to it goes out of scope.
+#define SC_CLEANUP(n) __attribute__((cleanup(n)))
```

Comment:
Do I guess someone suggested this at the sprint? :-)

Reply:
Not sure who you have in mind ;) But I added this, I find it (slightly) easier to read (but not much). I wish we could use __attribute__(cleanup(f)) to emulate a go defer. I think with some #define magic that is possible, however I am too lazy to look into it :)

Reply:
I'll have a look. I think Tyler asked me a while to do this but I never got to it. With this macro in place I'm more than willing to make the code uniformly using it.